test/runtime_cast_test.cpp

```diff
@@ -0,0 +1,239 @@
+//
+// Copyright Chris Glover, 2016.
+//
+// Distributed under the Boost Software License, Version 1.0. (See
+// accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#include <boost/type_index/runtime_cast.hpp>
+
+#include <boost/core/lightweight_test.hpp>
+
+
+// Classes include a member variable "name" with the
+// name of the class hard coded so we can be sure that
+// the pointer offsets are all working, since we're doing
+// a cast from void* at some point.
+
+#define IMPLEMENT_CLASS(type_name) \
+        type_name() : name( #type_name ) {} \
+        std::string name;
+
+struct base {
+    BOOST_TYPE_INDEX_REGISTER_CLASS_RTTI
```

Comment:
I can think of a few ways to get rid of this macro, and only have one, but to do so would, I think, require the user listing the name of the class something like;
`struct derived : base1, base2 {
  BOOST_TYPE_INDEX_REGISTER_CLASS_RTTI((derived)(base1)(base2))
}`
I'm not sure if that's preferable to having 2 macros or not.